lib/install/mutate-into-logical-tree.js

```diff
@@ -70,18 +70,27 @@ function translateTree_ (tree, seen) {
   tree.children.forEach(function (child) {
     pkg.dependencies[moduleName(child)] = translateTree_(child, seen)
   })
-  Object.keys(tree.missingDeps).forEach(function (name) {
+
+  function markMissing (name, requiredBy) {
     if (pkg.dependencies[name]) {
+      if (pkg.dependencies[name].missing) return
```

Comment:
`translateTree` is used to convert a npm@3 type tree into a `read-installed` type tree, so it's all based around the semantics that `read-installed` provides. It _is_ confusing. =/